src/runtime_src/core/common/api/xrt_hw_context.cpp

```diff
@@ -48,6 +49,17 @@ class hw_context_impl
     , m_hdl{m_core_device->create_hw_context(xclbin_id, m_cfg_param, m_mode)}
   {}
 
+  std::shared_ptr<hw_context_impl>
+  get_shared_ptr()
+  {
+    return shared_from_this();
+  }
+
+  ~hw_context_impl()
+  {
+    xrt_core::xdp::end_poll(this);
```

Comment:
Comment the asymmetry here. Explain that xrt::hw_context constructor started the polling and why polling was not started from the hw_context_impl constructor (shared ptr not available).   It isn't obvious, so it deserves a comment.